### PR TITLE
Typehinted Aggregation namespace

### PR DESCRIPTION
**Pull request by massimilianobraglia**
See https://github.com/ruflin/Elastica/issues/1548

### CODE REVIEW

**Comment by ruflin** [comment]
@massimilianobraglia For my understanding: This is now done manually or fully automatic?

**Comment by massimilianobraglia** [comment]
I'm updating manually

**Comment by ruflin** [comment]
Should php-cs fixer 2 allow to do this automatically?  I think I miss something.

**Comment by massimilianobraglia** [comment]
You're welcome!

File: lib/Elastica/Aggregation/AvgBucket.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function setBucketsPath($bucketsPath)
      *
      * @return $this
      */
-    public function setGapPolicy($gapPolicy)
+    public function setGapPolicy(string $gapPolicy): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/AvgBucket.php]: default value is ```skip```

File: lib/Elastica/Aggregation/AvgBucket.php [hunk 2]
```diff
@@ -54,7 +54,7 @@ public function setGapPolicy($gapPolicy)
      *
      * @return $this
      */
-    public function setFormat($format)
+    public function setFormat(string $format): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/AvgBucket.php]: ```null``` default value

File: lib/Elastica/Aggregation/BucketScript.php [hunk 3]
```diff
@@ -71,7 +71,7 @@ public function setGapPolicy($gapPolicy)
      *
      * @return $this
      */
-    public function setFormat($format)
+    public function setFormat(string $format): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/BucketScript.php]: gap_policy and format have their own default values

File: lib/Elastica/Aggregation/Cardinality.php [hunk 4]
```diff
@@ -13,12 +13,8 @@ class Cardinality extends AbstractSimpleAggregation
      *
      * @return $this
      */
-    public function setPrecisionThreshold($precisionThreshold)
+    public function setPrecisionThreshold(int $precisionThreshold): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/Cardinality.php]: cardinality has its own default value

File: lib/Elastica/Aggregation/DateHistogram.php [hunk 5]
```diff
@@ -11,11 +11,11 @@ class DateHistogram extends Histogram
     /**
      * Set time_zone option.
      *
-     * @param  string
+     * @param string $timezone
      *
      * @return $this
      */
-    public function setTimezone($timezone)
+    public function setTimezone(string $timezone): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/DateHistogram.php]: default timezone --> UTC

File: lib/Elastica/Aggregation/GeoDistance.php [hunk 6]
```diff
@@ -96,7 +96,7 @@ public function setUnit($unit)
      *
      * @return $this
      */
-    public function setDistanceType($distanceType)
+    public function setDistanceType(string $distanceType): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/GeoDistance.php]: default value ```arc```

File: lib/Elastica/Aggregation/GeoDistance.php [hunk 7]
```diff
@@ -84,7 +84,7 @@ public function addRange($fromValue = null, $toValue = null)
      *
      * @return $this
      */
-    public function setUnit($unit)
+    public function setUnit(string $unit): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/GeoDistance.php]: default value ```m```

File: lib/Elastica/Aggregation/GeohashGrid.php [hunk 8]
```diff
@@ -37,7 +37,7 @@ public function setField($field)
      *
      * @return $this
      */
-    public function setPrecision($precision)
+    public function setPrecision(int $precision): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/GeohashGrid.php]: defaults to 5

File: lib/Elastica/Aggregation/GeohashGrid.php [hunk 9]
```diff
@@ -49,7 +49,7 @@ public function setPrecision($precision)
      *
      * @return $this
      */
-    public function setSize($size)
+    public function setSize(int $size): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/GeohashGrid.php]: dafault to ```10,000```

File: lib/Elastica/Aggregation/SerialDiff.php [hunk 10]
```diff
@@ -54,7 +54,7 @@ public function setLag($lag)
      *
      * @return $this
      */
-    public function setGapPolicy($gapPolicy)
+    public function setGapPolicy(string $gapPolicy): self
```
Comment (p365labs) [inline lib/Elastica/Aggregation/SerialDiff.php]: default value for gap_policy

**Comment by massimilianobraglia** [comment]
@p365labs inserted the default values! Can you give it a look?

**Comment by massimilianobraglia** [comment]
@p365labs ready to merge! 🚀

**Pull request merged**